ref/Microsoft.Build/net/Microsoft.Build.cs

```diff
@@ -2054,3 +2054,20 @@ public partial struct XmlAttributeLink
         public string Value { get { throw null; } }
     }
 }
+namespace Microsoft.Build.Utilities
+{
+    public partial class ChangeWaves
```

Comment:
I switched it over to internal, and now I think it needs to be back at public. The latest addition to `TestEnvironment` (Making a `SetChangeWave` function) means that it needs to set `ChangeWaves.DisabledWave` to null, which it can't do if the class is anything less than public.

Reply:
What if you did that in ResetInstance_ForUnitTestsOnly instead? I would expect that method to reset DisabledWave anyway, and they're both in Microsoft.Build.Framework.UnitTests.csproj. It would also clean up the tests somewhat.